src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,51 @@
+import { useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { ChangeEvent } from 'react';
+import { SearchLink } from './SearchLink';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const centuriesList = ['16', '17', '18', '19', '20'];
+
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query') || '';
+
+  function handleSearch(e: ChangeEvent<HTMLInputElement>): void {
+    const newParam = (e.target.value === '')
+      ? getSearchWith(searchParams, { query: null })
+      : getSearchWith(searchParams, { query: e.target.value });
```

Comment:
```suggestion
      getSearchWith(searchParams, { query: e.target.value || null });
```

should work same but simplify